### PR TITLE
Tensorflow Philosophy blogpost

### PR DESCRIPTION
**Pull request by Rocketknight1**
Hey all! Now that the examples are refreshed, I did a quick pass over the blogpost and it should be almost ready. I just need to make a thumbnail to go with it and double-check my formatting everywhere.

### CODE REVIEW

File: tensorflow-philosophy.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            <span class="fullname">Matthew Carrigan</span>
+        </div>
+    </a>
+</div>
+
+
+### Introduction
+
+
+Despite increasing competition from PyTorch and JAX, TensorFlow remains [the most-used deep learning framework](https://twitter.com/fchollet/status/1478404084881190912?lang=en). It also differs from those other two libraries in some very important ways. In particular, it’s quite tightly integrated with its high-level API `Keras`, and its data loading library `tf.data`.
```
Comment (sgugger) [inline tensorflow-philosophy.md:31]: Lol, this sets the tone!

File: tensorflow-philosophy.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+As refined, respected and highly attractive TensorFlow engineers, we want to use the incredible power and flexibility of cutting-edge models, but we want to handle them with the tools and API we’re familiar with. This blogpost will be about the choices we make at Hugging Face to enable that, and what to expect from the framework as a TensorFlow programmer.
+
+### Interlude: 30 Seconds to :huggingface:
+
+Experienced users can feel free to skim or skip this section, but if this is your first encounter with Hugging Face and `transformers`, I should start by giving you an overview of the core idea of the library: You just ask for a pretrained model by name, and you get it in one line of code. The easiest way is to just use the `TFAutoModel` class:
+
+```py
+from transformers import TFAutoModel
+
+model = TFAutoModel.from_pretrained('bert-base-cased')
```
Comment (sgugger) [inline tensorflow-philosophy.md:44]:
```suggestion
model = TFAutoModel.from_pretrained("bert-base-cased")
```

File: tensorflow-philosophy.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+In Keras, the standard way to train a model is to create it, then `compile()` it with an optimizer and loss function, and finally `fit()` it. It’s very easy to load a model with transformers, but setting the loss function can be tricky - even for standard language model training, your loss function can be surprisingly non-obvious, and some hybrid models have extremely complex losses.
+
+Our solution to that is simple: If you `compile()` without a loss argument, we’ll give you the one you probably wanted. Specifically, we’ll give you one that matches both your base model and output type - if you `compile()` a BERT-based masked language model without a loss, we’ll give you a masked language modelling loss that handles padding and masking correctly, and will only compute losses on corrupted tokens, exactly matching the original BERT training process. If for some reason you really, really don’t want your model to be compiled with any loss at all, then simply specify `loss=None` when compiling.
+
+```py
+model = TFAutoModelForQuestionAnswering.from_pretrained('bert-base-cased')
+model.compile(optimizer='adam')  # No loss argument!
```
Comment (sgugger) [inline tensorflow-philosophy.md:120]:
```suggestion
model = TFAutoModelForQuestionAnswering.from_pretrained("bert-base-cased")
model.compile(optimizer="adam")  # No loss argument!
```

File: tensorflow-philosophy.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+dataset = load_dataset("glue", "cola")  # Simple text classification dataset
+dataset = dataset["train"]  # Just take the training split for now
+
+# Load our tokenizer and tokenize our data
+tokenizer = AutoTokenizer.from_pretrained('bert-base-cased')
+tokenized_data = tokenizer(dataset["text"], return_tensors="np", padding=True)
+labels = np.array(dataset["label"]) # Label is already an array of 0 and 1
+
+# Load and compile our model
+model = TFAutoModelForSequenceClassification.from_pretrained('bert-base-cased')
```
Comment (sgugger) [inline tensorflow-philosophy.md:154]:
```suggestion
model = TFAutoModelForSequenceClassification.from_pretrained("bert-base-cased")
```

File: tensorflow-philosophy.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+#### Philosophy #5: XLA is great!
+
+[XLA](https://www.tensorflow.org/xla) is the just-in-time compiler shared by TensorFlow and JAX. It converts linear algebra code into more optimized versions that run quicker and use less memory. It’s really cool and we try to make sure that we support it as much as possible. It’s extremely important for allowing models to be run on TPU, but it offers speed boosts for GPU and even CPU as well! To use it, simply `compile()` your model with the `jit_compile=True` argument (this works for all Keras models, not just Hugging Face ones):
+
+```py
+model.compile(optimizer='adam', jit_compile=True)
```
Comment (sgugger) [inline tensorflow-philosophy.md:191]:
```suggestion
model.compile(optimizer="adam", jit_compile=True)
```

File: tensorflow-philosophy.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+    def call(self, inputs):
+        tokenized = self.tokenizer(inputs)
+        return self.model(**tokenized)
+
+model = EndToEndModel(checkpoint='bert-base-cased')
+
+test_inputs = [
+    'This is a test sentence!',
+    'This is another one!'
```
Comment (sgugger) [inline tensorflow-philosophy.md:228]:
```suggestion
    "This is a test sentence!",
    "This is another one!",
```

File: tensorflow-philosophy.md (hunk 1 above)
Comment (Rocketknight1) [reply tensorflow-philosophy.md:31]: It's revenge for the :bell: tweet

**Comment by gante** [comment]
🔥
